3 - Assignment/240507 - Rust Basic + Roguelike Game, Week 3/generic-min/src/lib.rs

```diff
@@ -1,6 +1,13 @@
 use std::cmp::Ordering;
 
 // TODO: implement the `min` function.
+fn min<T: Ord>(v1: T, v2: T) -> T {
+    if v1.cmp(&v2) == Ordering::Less {
+        v1
+    } else {
+        v2
+    }
```

Comment:
열거체 타입의 경우 `if`문 보다는 `match`문을 이용한 패턴 매칭을 사용하는 게 좀 더 좋은 방식입니다.
```suggestion
    match v1.cmp(&v2) {
        Ordering::Less | Ordering::Equal => left,
        Ordering::Greater => right,
    }
```